### DISCUSSION TITLE
페이지네이션 시 중복 오류 해결 방안

### DISCUSSION THREAD

**Discussion by lina1919** [SWM-Space-Odyssey/WeddingMate_BackEnd#78] [General]
## 페이지네이션 시 중복 오류 발생
- 현재 포트폴리오 탐색 페이지에서 유저별로 랜덤한 사진들을 보여주려고 하다보니 pagination 호출 시 사진 데이터들이 중복되는 오류가 발생합니다.

## 예상 해결방안
1. 커서 기반 페이지네이션 
[https://velog.io/@leejh3224/%ED%8E%98%EC%9D%B4%EC%A7%80%EB%84%A4%EC%9D%B4%EC%85%98-%EC%BB%A4%EC%84%9C%EA%B8%B0%EB%B0%98-%ED%8E%98%EC%9D%B4%EC%A7%80%EA%B8%B0%EB%B0%98](url)
   - 하지만 이 경우 timeline 으로 보여지는 data에만 유효할 것으로 판단됨

2. db에 payload 칼럼을 만들고 그 안에 json 데이터로 user 별로 다른 아이디값 부여
3. 그냥 모든 유저에게 같은 리스트 보여주기

**Comment by lina1919** [comment]
개인적으로 엑스퍼트님께서 말해주신 유저별로 다른 데이터를 보여줘야할 필요성이 없다는 말에 공감하지만, 또 한번 생각해보면 저희 페이지는 '탐색' 기능을 수행하여 다양한 포트폴리오와 아이템을 찾을 수 있어야 할 것 같습니다. 
그 포트폴리오와 아이템을 다시 돌아왔을 때 찾을 수 없다는 것에 대해서는 좋아요 기능을 추가하여 나중에 다시 보게 하면 해결이 될 문제 같습니다. 
의견 부탁드립니다! @nitronium102 @CSKIM999

**Comment by nitronium102** [comment]
좋아요로 저장하는 방식은 인스타랑 비슷하네요! 저는 개인적으로 1번이랑 좋아요를 합치면 좋을 것 같습니다!
커서 기반 페이지네이션을 통해 보여주되 아이템 id보다 max_id가 넘어가면 max_id를 0으로 초기화시켜 처음부터 다시 보여주는 형식을 사용하고, 여기에 좋아요 기능을 추가하여 사용자가 원하는 아이템이나 포트폴리오를 바로 찾을 수 있게 하면 좋을 것 같아요
